src/model/Space.hpp

```diff
@@ -640,6 +641,30 @@ namespace model {
     /** Returns all ZoneMixing objects which exhaust air from this space */
     std::vector<ZoneMixing> exhaustZoneMixing() const;
 
+    // Find all surfaces where the outwardNormal does not point towards the outside of the Space
+    std::vector<Surface> findSurfacesWithIncorrectOrientation() const;
+
+    // Checks the outwardNormal of every surface points towards the outside of the Space
+    bool areAllSurfacesCorrectlyOriented() const;
+
+    // Returns true if the orientation of any surface has been changed
+    bool fixSurfacesWithIncorrectOrientation();
+
+    /** This method will check the floorPrint of the space, if that can't be computed, it's not convex. If it can, it checks whether that resulting
+        * Surface3d is convex. Note: having a floorPrint that's convex isn't sufficied to deem a space to be convex, the walls could still be non
+        * convex, but it should suit most typical applications */
+    bool isConvex() const;
+
+    /** Checks if every Surface is convex, returns the Concave ones.
+        * Note: having a non convex surface does not necesarilly mean that the Space is not convex
+        * eg: a box with a wall that is split into two L s would return false, while the Space is actually still convex. */
+    std::vector<Surface> findNonConvexSurfaces() const;
+
+    /// @cond
+    void cacheGeometryDiagnostics();
+    void resetCachedGeometryDiagnostics();
+    /// @endcond
```

Comment:
Hidden methods to cache... I can't very reliably use signals onChange because it depends on the surfaces... so I'm choosing not to, but making this hidden, so the ThreeJSForwardTranslator can use it.